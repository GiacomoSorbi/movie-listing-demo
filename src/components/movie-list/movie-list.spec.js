import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { movies, genres } from '../../fixtures'

import MovieList from './'

describe('MovieList should', () => {
  it('exist', () => {
    expect(MovieList).toBeDefined()
  })

  it('have a consistent structure when displaying movies', () => {
    const wrapper = mount(
      <MovieList
        filters={[12]}
        genres={genres}
        onChangeGenreFilters={jest.fn()}
        onChangeMinRating={jest.fn()}
        movies={movies}
        minRating='7'
      />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('have a consistent structure when not displaying movies (filtered out)', () => {
    const wrapper = mount(
      <MovieList
        filters={[15]}
        genres={genres}
        onChangeGenreFilters={jest.fn()}
        onChangeMinRating={jest.fn()}
        movies={movies}
        minRating='10'
      />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
