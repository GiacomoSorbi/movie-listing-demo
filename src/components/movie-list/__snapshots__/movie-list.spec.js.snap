// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should have a consistent structure when displaying movies 1`] = `
<MovieList
  filters={
    Array [
      12,
    ]
  }
  genres={
    Object {
      "12": "Feline Action",
      "15": "Feline Formation Novel",
      "18": "Feline Adventure",
      "19432": "Feline Philosophy",
      "23": "Feline Romance",
      "24": "Feline Western",
      "26": "Feline Drama",
    }
  }
  minRating="7"
  movies={
    Array [
      Object {
        "genre_ids": Array [
          12,
          15,
          26,
          19432,
        ],
        "original_title": "A Feline Adventure",
        "overview": "Pippi is a cat on a quest to sleep more",
        "poster_path": "/pippi.jpg",
        "title": "Pippi the Cat",
        "vote_average": 9,
      },
      Object {
        "genre_ids": Array [
          12,
          18,
          24,
          26,
          19432,
        ],
        "original_title": "the Revenge of Fluffy",
        "overview": "A Daring Fight for the Last Straw of Catnip",
        "poster_path": "/pippi2.jpg",
        "title": "Pippi the Cat II",
        "vote_average": 7.5,
      },
      Object {
        "genre_ids": Array [
          12,
          18,
          23,
          26,
          19432,
        ],
        "original_title": "Pippi the Cat III",
        "overview": "Pippi Sleeps for 2 Hours and More!",
        "poster_path": "/pippi3.jpg",
        "title": "Pippi the Cat III",
        "vote_average": 10,
      },
    ]
  }
  onChangeGenreFilters={[MockFunction]}
  onChangeMinRating={[MockFunction]}
>
  <section
    className="movie-list"
  >
    <MovieCard
      genres={
        Object {
          "12": "Feline Action",
          "15": "Feline Formation Novel",
          "18": "Feline Adventure",
          "19432": "Feline Philosophy",
          "23": "Feline Romance",
          "24": "Feline Western",
          "26": "Feline Drama",
        }
      }
      key="Pippi the Cat"
      movie={
        Object {
          "genre_ids": Array [
            12,
            15,
            26,
            19432,
          ],
          "original_title": "A Feline Adventure",
          "overview": "Pippi is a cat on a quest to sleep more",
          "poster_path": "/pippi.jpg",
          "title": "Pippi the Cat",
          "vote_average": 9,
        }
      }
    >
      <div
        className="movie-card"
      >
        <div
          className="movie-card-content"
        >
          <img
            alt="Pippi the Cat"
            className="movie-poster"
            src="http://image.tmdb.org/t/p/w300/pippi.jpg"
          />
          <div
            className="movie-info"
          >
            <h3>
              Pippi the Cat (Original Title: A Feline Adventure)
            </h3>
            <div
              className="movie-genres"
            >
              <h4>
                Genres:
              </h4>
              <ul>
                <li
                  className="genre-item"
                  key="12"
                >
                  Feline Action
                </li>
                <li
                  className="genre-item"
                  key="15"
                >
                  Feline Formation Novel
                </li>
                <li
                  className="genre-item"
                  key="26"
                >
                  Feline Drama
                </li>
                <li
                  className="genre-item"
                  key="19432"
                >
                  Feline Philosophy
                </li>
              </ul>
            </div>
            <div
              className="movie-description desktop-only"
            >
              <h4>
                Description:
              </h4>
              <p>
                Pippi is a cat on a quest to sleep more
              </p>
            </div>
          </div>
        </div>
      </div>
    </MovieCard>
    <MovieCard
      genres={
        Object {
          "12": "Feline Action",
          "15": "Feline Formation Novel",
          "18": "Feline Adventure",
          "19432": "Feline Philosophy",
          "23": "Feline Romance",
          "24": "Feline Western",
          "26": "Feline Drama",
        }
      }
      key="Pippi the Cat II"
      movie={
        Object {
          "genre_ids": Array [
            12,
            18,
            24,
            26,
            19432,
          ],
          "original_title": "the Revenge of Fluffy",
          "overview": "A Daring Fight for the Last Straw of Catnip",
          "poster_path": "/pippi2.jpg",
          "title": "Pippi the Cat II",
          "vote_average": 7.5,
        }
      }
    >
      <div
        className="movie-card"
      >
        <div
          className="movie-card-content"
        >
          <img
            alt="Pippi the Cat II"
            className="movie-poster"
            src="http://image.tmdb.org/t/p/w300/pippi2.jpg"
          />
          <div
            className="movie-info"
          >
            <h3>
              Pippi the Cat II (Original Title: the Revenge of Fluffy)
            </h3>
            <div
              className="movie-genres"
            >
              <h4>
                Genres:
              </h4>
              <ul>
                <li
                  className="genre-item"
                  key="12"
                >
                  Feline Action
                </li>
                <li
                  className="genre-item"
                  key="18"
                >
                  Feline Adventure
                </li>
                <li
                  className="genre-item"
                  key="24"
                >
                  Feline Western
                </li>
                <li
                  className="genre-item"
                  key="26"
                >
                  Feline Drama
                </li>
                <li
                  className="genre-item"
                  key="19432"
                >
                  Feline Philosophy
                </li>
              </ul>
            </div>
            <div
              className="movie-description desktop-only"
            >
              <h4>
                Description:
              </h4>
              <p>
                A Daring Fight for the Last Straw of Catnip
              </p>
            </div>
          </div>
        </div>
      </div>
    </MovieCard>
    <MovieCard
      genres={
        Object {
          "12": "Feline Action",
          "15": "Feline Formation Novel",
          "18": "Feline Adventure",
          "19432": "Feline Philosophy",
          "23": "Feline Romance",
          "24": "Feline Western",
          "26": "Feline Drama",
        }
      }
      key="Pippi the Cat III"
      movie={
        Object {
          "genre_ids": Array [
            12,
            18,
            23,
            26,
            19432,
          ],
          "original_title": "Pippi the Cat III",
          "overview": "Pippi Sleeps for 2 Hours and More!",
          "poster_path": "/pippi3.jpg",
          "title": "Pippi the Cat III",
          "vote_average": 10,
        }
      }
    >
      <div
        className="movie-card"
      >
        <div
          className="movie-card-content"
        >
          <img
            alt="Pippi the Cat III"
            className="movie-poster"
            src="http://image.tmdb.org/t/p/w300/pippi3.jpg"
          />
          <div
            className="movie-info"
          >
            <h3>
              Pippi the Cat III
            </h3>
            <div
              className="movie-genres"
            >
              <h4>
                Genres:
              </h4>
              <ul>
                <li
                  className="genre-item"
                  key="12"
                >
                  Feline Action
                </li>
                <li
                  className="genre-item"
                  key="18"
                >
                  Feline Adventure
                </li>
                <li
                  className="genre-item"
                  key="23"
                >
                  Feline Romance
                </li>
                <li
                  className="genre-item"
                  key="26"
                >
                  Feline Drama
                </li>
                <li
                  className="genre-item"
                  key="19432"
                >
                  Feline Philosophy
                </li>
              </ul>
            </div>
            <div
              className="movie-description desktop-only"
            >
              <h4>
                Description:
              </h4>
              <p>
                Pippi Sleeps for 2 Hours and More!
              </p>
            </div>
          </div>
        </div>
      </div>
    </MovieCard>
  </section>
</MovieList>
`;

exports[`MovieList should have a consistent structure when not displaying movies (filtered out) 1`] = `
<MovieList
  filters={
    Array [
      15,
    ]
  }
  genres={
    Object {
      "12": "Feline Action",
      "15": "Feline Formation Novel",
      "18": "Feline Adventure",
      "19432": "Feline Philosophy",
      "23": "Feline Romance",
      "24": "Feline Western",
      "26": "Feline Drama",
    }
  }
  minRating="10"
  movies={
    Array [
      Object {
        "genre_ids": Array [
          12,
          15,
          26,
          19432,
        ],
        "original_title": "A Feline Adventure",
        "overview": "Pippi is a cat on a quest to sleep more",
        "poster_path": "/pippi.jpg",
        "title": "Pippi the Cat",
        "vote_average": 9,
      },
      Object {
        "genre_ids": Array [
          12,
          18,
          24,
          26,
          19432,
        ],
        "original_title": "the Revenge of Fluffy",
        "overview": "A Daring Fight for the Last Straw of Catnip",
        "poster_path": "/pippi2.jpg",
        "title": "Pippi the Cat II",
        "vote_average": 7.5,
      },
      Object {
        "genre_ids": Array [
          12,
          18,
          23,
          26,
          19432,
        ],
        "original_title": "Pippi the Cat III",
        "overview": "Pippi Sleeps for 2 Hours and More!",
        "poster_path": "/pippi3.jpg",
        "title": "Pippi the Cat III",
        "vote_average": 10,
      },
    ]
  }
  onChangeGenreFilters={[MockFunction]}
  onChangeMinRating={[MockFunction]}
>
  <section
    className="movie-list"
  >
    <h3
      className="error"
    >
      No Movies Matching the Current Filter Criteria
    </h3>
  </section>
</MovieList>
`;
